agent/dns.go

```diff
@@ -717,6 +717,37 @@ func syncExtra(index map[string]dns.RR, resp *dns.Msg) {
 	resp.Extra = extra
 }
 
+// dnsBinaryTruncate find the optimal number of records using a fast binary search and return
+// it in order to return a DNS answer lower than maxSize parameter.
+func dnsBinaryTruncate(resp *dns.Msg, maxSize int, index map[string]dns.RR, hasExtra bool) int {
+	originalAnswser := resp.Answer
+	originalExtra := resp.Extra
+	originalIndex := index
+	startIndex := 0
+	endIndex := len(resp.Answer) + 1
+	for endIndex-startIndex > 1 {
+		median := startIndex + (endIndex-startIndex)/2
+
+		resp.Answer = originalAnswser[:median]
+		if hasExtra {
+			resp.Extra = originalExtra[:median]
```

Comment:
Correct me if you think otherwise but I think lines 733 and 734 here are unnecessary as well as the originalIndex and originalExtra variables.

The hasExtra block should just be a call to syncExtra. slicing originalExtra should not do anything (other than eat a few cpu cycles) as syncExtra will just overwrite resp.Extra with a brand new slice of its making. Also since syncExtra doesn't modify the RR index there shouldn't be a need to hold the originalIndex and reset index to it.

So I would remove those two vars and those extra lines

Reply:
@mkeeler Yes, you are right, I was confused by a unit test at a given time, but since no modification is performed, it is safe. Done in next patch